main.py

```diff
@@ -1,127 +1,201 @@
-        traceback.print_stack(sys._current_frames()[th.ident])
-        print("")
-signal.signal(signal.SIGUSR1, dumpthreads)
+LOGGING_PATH = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)),
+    'pylci.log'
+)
 
-#Getting pyLCI config, it will be passed to input and output initializers
-#If config at main_config_path exists, use that
-#If not, use the backup path
-#Also, log the errors to a file so that it can be debugged later
+LOGGING_FMT = (
+    '[%(levelname)s] %(asctime)s %(filename)s [%(process)d]: %(message)s',
+    '%Y-%m-%d %H:%M:%S'
+)
 
-from helpers import read_config
+CONFIG_PATHS = [
+    '/boot/pylci_config.json',
+    './config.json'
+]
+
+
+def init():
+    """
+    Initialize input and output objects
+    """
+
+    config = None
+
+    # Load config
+    for path in CONFIG_PATHS:
+        try:
+            logging.debug('Loading config from {0}'.format(path))
+            config = read_config(path)
+        except:
+            logging.exception('Failed to load config')
+        else:
+            logging.debug('Successfully loaded config')
+            break
 
-is_emulator = emulator_flag_filename in os.listdir(".")
+    if config is None:
+        sys.exit('Failed to load any config file')
+
+    # Initialize output
+    try:
+        output.init(config['output'])
+        o = output.screen
+    except:
+        logging.exception('Failed to initialize the output object')
+        raise
 
-if not is_emulator:
+    # Initialize input
     try:
-        error_file = open(config_error_file, "w+")
-        config = read_config(main_config_path)
-    except Exception as e:
-        print(repr(e))
-        print("------------------------------")
-        print("Couldn't read main config, using backup config!")
-        error_file.write("Couldn't read main config: {}\n".format(repr(e)))
-        error_file.write("Using backup config!\n")
+        # Now we can show errors on the display
+        input.init(config['input'])
+        i = input.listener
+    except:
+        logging.exception('Failed to initialize the input object')
+        Printer(['Oops. :(', 'y u make mistake'], None, o, 0)
+        raise
+
+    return i, o
+
+
+def launch(name=None, **kwargs):
+    """
+    Launches pyLCI, either in full mode or in
+    single-app mode (if ``name`` kwarg is passed).
+    """
+
+    i, o = init()
+    app_man = AppManager('apps', Menu, Printer, i, o)
+
+    if name is None:
         try:
-            config = read_config(backup_config_path)
-        except Exception as e:
-            print("Couldn't read backup config, exiting!")
-            error_file.write("Couldn't read backup config: {}\n".format(repr(e)))
-            error_file.write("Exiting!\n")
-            error_file.close()
-            sys.exit(1)
-    else:
-        if os.path.exists(config_error_file): os.remove(config_error_file)
-        error_file.close()
-else:
-    config = read_config(backup_config_path)
+            splash(i, o)
```

Comment:
Also, keep in mind that you're not using "from splash import splash" or just "splash.splash", so it seems that calling splash() like that will fail.